src/main/java/com/chattriggers/ctjs/mixins/CreativeInventoryScreenMixin.java

```diff
@@ -0,0 +1,40 @@
+package com.chattriggers.ctjs.mixins;
+
+import com.chattriggers.ctjs.minecraft.wrappers.Player;
+import com.chattriggers.ctjs.minecraft.wrappers.inventory.Item;
+import com.chattriggers.ctjs.triggers.TriggerType;
+import net.minecraft.client.gui.screen.ingame.CreativeInventoryScreen;
+import net.minecraft.item.ItemStack;
+import net.minecraft.screen.slot.Slot;
+import net.minecraft.screen.slot.SlotActionType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.Objects;
+
+@Mixin(CreativeInventoryScreen.class)
+public class CreativeInventoryScreenMixin {
+    @Inject(method = "onMouseClick", at = @At(value = "FIELD", target = "Lnet/minecraft/client/gui/screen/ingame/CreativeInventoryScreen;lastClickOutsideBounds:Z", shift = At.Shift.BY, by = 2), cancellable = true)
+    private void injectOnMouseClick(Slot slot, int slotId, int button, SlotActionType actionType, CallbackInfo ci) {
+        // dropping by clicking outside creative tab
+        TriggerType.DROP_ITEM.triggerAll(new Item(Objects.requireNonNull(Player.getPlayer()).currentScreenHandler.getCursorStack()), button == 0, ci);
+    }
+
+    @Inject(method = "onMouseClick", at = @At(value = "INVOKE", target = "Lnet/minecraft/screen/slot/Slot;takeStack(I)Lnet/minecraft/item/ItemStack;"), cancellable = true)
+    private void injectOnMouseClick1(Slot slot, int slotId, int button, SlotActionType actionType, CallbackInfo ci) {
+        // dropping item from slot in creative inventory
+        TriggerType.DROP_ITEM.triggerAll(new Item(slot.getStack()), button == 0, ci);
+    }
+
+    // TODO: This should be ordinal 1, but ordinal 0 is generating the correct injection point???
```

Comment:
I'm hesitant to merge this PR with this comment lol. Definitely seems like something we should figure out

Reply:
I am waiting for this comment to be resolved before merging this PR btw